version: '3.4'

services:

  rabbitmq:
    container_name: rabbitmq
    hostname: rabbitmq
    image: rabbitmq:latest
#    networks:
#      - main
    ports:
      - "5672:5672"
    restart: on-failure

  postgres:
    container_name: postgres
    hostname: postgres
    image: postgres:latest
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
      - POSTGRES_DB=postgres
#    networks:
#      - main
    ports:
      - "5432:5432"
    restart: on-failure
    volumes:
      - postgresql-data:/var/lib/postgresql/data

  pgadmin:
    container_name: pgadmin_container
    image: dpage/pgadmin4
    environment:
      PGADMIN_DEFAULT_EMAIL: fm.maghraoui@gmail.com
      PGADMIN_DEFAULT_PASSWORD: ${PGADMIN_DEFAULT_PASSWORD:-admin}
      PGADMIN_CONFIG_SERVER_MODE: 'False'
    volumes:
      - pgadmin:/var/lib/pgadmin

    ports:
      - "${PGADMIN_PORT:-5050}:80"
#    networks:
#      - main
    restart: unless-stopped

  app:
    build:
      context: .
      dockerfile: docker/dev/Dockerfile
    command: sh -c "python manage.py collectstatic --no-input && python manage.py migrate && gunicorn coreapp.wsgi -b 0.0.0.0:8000"
    container_name: app
    depends_on:
      - postgres
      - rabbitmq
    expose:
      - "8000"
    hostname: app
    image: app-image
#    networks:
#      - main
    restart: on-failure

  celery_worker:
    command: sh -c "celery -A coreapp worker -l info"
    container_name: celery_worker
    depends_on:
      - app
#      - postgres
#      - rabbitmq
    volumes:
      - .:/app
    deploy:
      replicas: 1
      restart_policy:
        condition: on-failure

    hostname: celery_worker
    image: app-image
#    networks:
#      - main
    restart: on-failure

  celery_beat:
    command: sh -c "celery -A coreapp beat -l info --scheduler django_celery_beat.schedulers:DatabaseScheduler"
    container_name: celery_beat
    depends_on:
      - app
#      - postgres
#      - rabbitmq
    volumes:
      - .:/app
    hostname: celery_beat
    image: app-image
#    networks:
#      - main
    restart: on-failure

#networks:
#  main:

volumes:
  postgresql-data:
  pgadmin:

networks:
  host:
    name: farid_maghraoui_challenge_network
    external: true